Cleartext communication not allowed on Android 9.0/Pie

Cleartext communication isn't allowed on Android Pie by default, so currently at least an SSL enabled Gotify server must be used.

Error message in adblog: `Caused by: java.net.UnknownServiceException: CLEARTEXT communication to <IP> not permitted by network security policy`

Tested by connecting to a default Gotify server.

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -13,6 +13,7 @@
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher"
         android:supportsRtl="true"
+        android:usesCleartextTraffic="true"
         android:theme="@style/AppTheme">
         <activity
             android:name=".init.InitializationActivity"

--- a/app/src/main/java/com/github/gotify/login/LoginActivity.java
+++ b/app/src/main/java/com/github/gotify/login/LoginActivity.java
@@ -102,9 +102,14 @@ public class LoginActivity extends AppCompatActivity {
     @OnClick(R.id.checkurl)
     public void doCheckUrl() {
         String url = urlField.getText().toString();
-        if (HttpUrl.parse(url) == null) {
+        HttpUrl parsedUrl = HttpUrl.parse(url);
+        if (parsedUrl == null) {
             Utils.showSnackBar(LoginActivity.this, "Invalid URL (include http:// or https://)");
             return;
+        }
+
+        if ("http".equals(parsedUrl.scheme())) {
+            showHttpWarning();
         }
 
         checkUrlProgress.setVisibility(View.VISIBLE);
@@ -115,6 +120,15 @@ public class LoginActivity extends AppCompatActivity {
         ClientFactory.versionApi(fixedUrl, tempSSLSettings())
                 .getVersion()
                 .enqueue(callInUI(this, onValidUrl(fixedUrl), onInvalidUrl(fixedUrl)));
+    }
+
+    public void showHttpWarning() {
+        new AlertDialog.Builder(this)
+                .setTitle(R.string.warning)
+                .setCancelable(true)
+                .setMessage(R.string.http_warning)
+                .setPositiveButton(R.string.i_understand, (a, b) -> {})
+                .show();
     }
 
     @OnClick(R.id.open_logs)

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -54,4 +54,7 @@
     <string name="done">Done</string>
     <string name="no_certificate_selected">No certificate selected</string>
     <string name="remove_ca_certificate">Remove CA Certificate</string>
+    <string name="warning">Warning</string>
+    <string name="http_warning">Using http is insecure and it\'s recommend to use https instead. Use your favorite search engine to get more information about this topic.</string>
+    <string name="i_understand">I Understand</string>
 </resources>

